examples/query_string.rs

```diff
@@ -0,0 +1,35 @@
+use serde_derive::{Deserialize, Serialize};
+use std::collections::HashMap;
+use warp::{http::Response, Filter};
+
+#[derive(Deserialize, Serialize)]
+struct MyObject {
+    key1: String,
+    key2: u32,
+}
+
+#[tokio::main]
+async fn main() {
+    pretty_env_logger::init();
+
+    // get /example1?key=value
+    let example1 = warp::get()
+        .and(warp::path("example1"))
+        .and(warp::query::<HashMap<String, String>>())
+        .map(|p: HashMap<String, String>| match p.get("key") {
+            Some(key) => Response::builder().body(format!("key = {}", key)),
+            None => Response::builder().body(String::from("No param in query.")),
```

Comment:
can we set the status to `BAD_REQUEST`?

Reply:
No problem at all. But, are we sure we want to consider the absence of a query param an error here? Some params are optional, which is why I did not consider it a bad request in the example.

Reply:
As an alternative, maybe the text should say "key = <missing>" instead.

Just a thought. Let me know and I'll change it, I have no strong opinion either way.

Reply:
> Some params are optional, which is why I did not consider it a bad request in the example.

ah i see! can we then leave a comment that its optional?
btw, can we also add a third example showcasing how to deal with absent/failed to decode query parms, like:
```Rust
    let opt_query = warp::query::<MyObject>()
        .map(Some)
        .or_else(|_| async {
            Ok::<(Option<MyObject>,), std::convert::Infallible>((None,))
        });

    // get /example3?key1=value,key2=42
    let example3 = warp::get()
        .and(warp::path("example3"))
        .and(opt_query)
        .map(|p: Option<MyObject>| {
            match p {
                Some(obj) => Response::builder().body(format!("key1 = {}, key2 = {}", obj.key1, obj.key2)),
                None => Response::builder()
                .status(StatusCode::BAD_REQUEST)
                .body(String::from("failed to decode query param")),
            }
        });
```

Reply:
Good plan. I pushed some changes, including your example3. I formatted it a bit so it resembles example two, to make it more obvious to readers that it builds on example2 with custom error handling. I also added comments.

I hope you agree that it makes it clearer. If not, just let me know :)